### ISSUE TITLE
Bug - uibDropdown + ngTouch - not closing after touch

### ISSUE THREAD

**Issue by radarfox** [angular-ui/bootstrap#5101]
I have found this bug in the dropdown module and created plunker to demonstrate it: http://plnkr.co/edit/cQ8zjbqFfYQ4KMlI4raJ

Conditions:

* angular v1.4.8
* angular-touch v1.4.8
* angular-bootstrap v0.14.3
* touch device or Chrome with "Emulate touch screen" option
* make a "uib-dropdown-menu" with anchor elements using ngClick directive

Result:

Dropdown menu is not closing after touch on any of its item with ngClick. If you exclude ngTouch or do not use ngClick directive everything works ok. Also clicking outside menu works as expected.

**Comment by wesleycho** [comment]
Thanks a lot for this with detailed reproduction - one work around for now is to have your ng-click in each item set the `isOpen` to `false`.

This may be tricky to fix, but I'm in the process of trying to figure out a good fix.

**Comment by wesleycho** [comment]
Took a look further, and I don't think there is anything we can do in the library itself - the problem here is that with ngTouch, ng-click is overrided with a custom version that calls `event.stopPropagation` on the event, as seen in https://github.com/angular/angular.js/blob/74eb17d7c8232f72f134bf2546f10fed7234d276/src/ngTouch/directive/ngClick.js#L159 .

I will accept updating of the documentation to mention this workaround for when one uses ngTouch though.

**Comment by wesleycho** [comment]
cc @Foxandxss

**Comment by radarfox** [comment]
Proposed work around should work, but it is not very handy to have to care about each action of each dropdown to make sure dropdown gets closed.

Are you sure, that the problem is in ngTouch? I see plenty of `return` before `event.stopPropagation`. That code should only prevent unwanted clicks. Also this function is called on non-touch devices and there is no bug.

I think that here is part of code where event is created, but somehow it does not trigger dropdown close https://github.com/angular/angular.js/blob/74eb17d7c8232f72f134bf2546f10fed7234d276/src/ngTouch/directive/ngClick.js#L242

**Comment by wesleycho** [comment]
I did some experimentation console.logging, the click callbacks never gets fired because the ng-click swallows the event.

**Comment by radarfox** [comment]
ngClick is supposed to prevent the original `click` event, but it should emit fake `click` just after `touchend` - without having to wait 200ms or so on touch devices.

**Comment by Foxandxss** [comment]
I have a fix underway. Working on the tests.

**Comment by radarfox** [comment]
I have updated plunker and it seems that click events are really blocked by ngTouch http://plnkr.co/edit/cQ8zjbqFfYQ4KMlI4raJ

**Comment by Foxandxss** [comment]
So I tried to make a workaround for this in our end, but it introduces more issues that the ones it fixes.

So unless someone proves that I am wrong, I don't think we can do too much in here :(

**Comment by wesleycho** [comment]
I think we just should update the docs with a workaround for this situation. This will prove too difficult to fix in the library itself without very subpar code IMO.

**Comment by anyandallart** [comment]
I encountered another random issue (not sure what caused it) where ngTouch was interfering with my uib-dropdown. I created a directive modified after the angular UI dropdown that seems to work with touch devices and achieves the basic functionality. 

If anyone has any suggestions on any issues with or improvements to the code it would be much appreciated.

https://plnkr.co/edit/7zIFO4pTS93pqk8UH6KK?p=preview

**Comment by icfantv** [comment]
@anyandallart, please do not comment on closed issues and ask for help.  Per our project documentation, please hit up StackOverflow for support-related questions or if you think you've found a bug, please enter a new issue along with a minimally working plunker to show the problem.

**Comment by anyandallart** [comment]
@icfantv no argument here, just clarifying my comment. It was less of a request for help and more of a suggestion on how I handled touch events to get the angular dropdown (using snippets from angular ui) and ngtouch to work together without complication. I've benefited a lot from your code, so I wanted to contribute my workaround for other users who stumbled upon this thread like I did. It undoubtedly could be optimized and might even have serious flaws, so just throwing it out there. If there is a better place for it I'm happy to put it elsewhere. Thanks for all of your work on this.
